### PR TITLE
[Flow Decomposition] Update to Entsoe 2.1.0

### PR DESCRIPTION
**Pull request by OpenSuze**
Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by OpenSuze** [comment]
Current new features:
- Add contingency management
- Add additional xnec provider support with union provider

**Comment by OpenSuze** [comment]
New feature:
- Not merged X node flow decomposition

File: cpp/src/bindings.cpp [hunk 1]
```diff
@@ -625,8 +625,17 @@ PYBIND11_MODULE(_pypowsybl, m) {
     
     m.def("create_flow_decomposition", &pypowsybl::createFlowDecomposition, "Create a security analysis");
 
-    m.def("add_precontingency_monitored_elements_for_flow_decomposition", &pypowsybl::addPrecontingencyMonitoredElementsForFlowDecomposition, "Add elements to be monitored for a flow decomposition",
-          py::arg("flow_decomposition_context"), py::arg("elements_ids"));
+    m.def("add_contingency_for_flow_decomposition", &pypowsybl::addContingencyForFlowDecomposition, "Add a contingency for flow decomposition",
```
Comment (EtienneLt) [inline cpp/src/bindings.cpp:634]: Is it needed to specified "for flow decomposition" ? Is it in the module with the same name ? what do you think ?

File: cpp/src/pypowsybl.h [hunk 2]
```diff
@@ -132,9 +132,10 @@ enum ConnectedComponentMode {
     ALL,
 };
 
-enum XnecSelectionStrategy {
-    ONLY_INTERCONNECTIONS = 0,
-    INTERCONNECTION_OR_ZONE_TO_ZONE_PTDF_GT_5PC,
+enum DefaultXnecProvider {
+    GT_5_PERC_ZONE_TO_ZONE_PTDF = 0,
+    ALL_BRANCHES,
+    INTERCONNECTIONS,
```
Comment (EtienneLt) [inline cpp/src/pypowsybl.h:138]: Sure about that ? is it a breaking change ?

File: java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionCFunctions.java [hunk 3]
```diff
@@ -98,7 +134,7 @@ public static PyPowsyblApiHeader.FlowDecompositionParametersPointer createFlowDe
 
     @CEntryPoint(name = "freeFlowDecompositionParameters")
     public static void freeFlowDecompositionParameters(IsolateThread thread, PyPowsyblApiHeader.FlowDecompositionParametersPointer flowDecompositionParametersPtr,
-                                              PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+                                                       PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionCFunctions.java:140]: a bit strange

File: cpp/src/bindings.cpp (hunk 1 above)
Comment (EtienneLt) [reply cpp/src/bindings.cpp:634]: ok it is in _pypowsybl so it is needed

File: pypowsybl/flowdecomposition.py [hunk 4]
```diff
@@ -97,21 +104,122 @@ def __repr__(self) -> str:
...
+            contingency_id = '_'.join(elements_ids)
+        _pypowsybl.add_contingency_for_flow_decomposition(self._handle, contingency_id=contingency_id, elements_ids=elements_ids)
+        return self
+
+    def add_monitored_elements(self, branch_ids: _Union[_List[str], str],
+                               contingency_ids: _Union[_List[str], str] = None,
+                               contingency_context_type: ContingencyContextType = ContingencyContextType.ALL) -> FlowDecomposition:
         """
-        Add elements to be monitored by the flow decomposition on precontingency state.
+        Add branches to be monitored by the flow decomposition.
```
Comment (EtienneLt) [inline pypowsybl/flowdecomposition.py:156]: should it be renamed add_monitored_branches ?

File: cpp/src/pypowsybl.h (hunk 2 above)
Comment (OpenSuze) [reply cpp/src/pypowsybl.h:138]: Yes, I forgot to remove those in #536. This PR contains breaking changes

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**